GNU coreutils has '-c' option for formatted output, not -f.
Don't spam stderr when directory gets created.
diff --git a/scripts/nix-profile-daemon.sh.in b/scripts/nix-profile-daemon.sh.in
index 7f46cc6..386f607 100644
--- a/scripts/nix-profile-daemon.sh.in
+++ b/scripts/nix-profile-daemon.sh.in
@@ -15,3 +15,3 @@ export NIX_PROFILES="@localstatedir@/nix/profiles/default $HOME/.nix-profile"
 mkdir -m 0755 -p $NIX_USER_PROFILE_DIR
-if test "$(stat -f '%u' $NIX_USER_PROFILE_DIR)" != "$(id -u)"; then
+if test "$(stat -c '%u' $NIX_USER_PROFILE_DIR)" != "$(id -u)"; then
     echo "WARNING: bad ownership on $NIX_USER_PROFILE_DIR" >&2
@@ -37,3 +37,3 @@ if test -w $HOME; then
   mkdir -m 0755 -p $NIX_USER_GCROOTS_DIR
-  if test "$(stat -f '%u' $NIX_USER_GCROOTS_DIR)" != "$(id -u)"; then
+  if test "$(stat -c '%u' $NIX_USER_GCROOTS_DIR)" != "$(id -u)"; then
       echo "WARNING: bad ownership on $NIX_USER_GCROOTS_DIR" >&2
diff --git a/scripts/nix-profile.sh.in b/scripts/nix-profile.sh.in
index 4f09b55..e93ef42 100644
--- a/scripts/nix-profile.sh.in
+++ b/scripts/nix-profile.sh.in
@@ -5,3 +5,2 @@ if [ -n "$HOME" ]; then
     if ! [ -L "$NIX_LINK" ]; then
-        echo "creating $NIX_LINK" >&2
         _NIX_DEF_LINK=@localstatedir@/nix/profiles/per-user/$USER/profile
